src/server/scripts/Kalimdor/RuinsOfAhnQiraj/boss_kurinnaxx.cpp

```diff
@@ -82,10 +82,15 @@ struct boss_kurinnaxx : public BossAI
         if (killer)
         {
             killer->GetMap()->LoadGrid(-9502.80f, 2042.65f); // Ossirian grid
+            killer->GetMap()->LoadGrid(-8538.17f, 1486.09f); // Andorov run path grid
 
             if (Player* player = killer->GetCharmerOrOwnerPlayerOrPlayerItself())
             {
-                player->SummonCreature(NPC_ANDOROV, -8538.177f, 1486.0956f, 32.39054f, 3.7638654f, TEMPSUMMON_CORPSE_DESPAWN, 600000000);
+                Creature* creature = player->SummonCreature(NPC_ANDOROV, -8538.177f, 1486.0956f, 32.39054f, 3.7638654f, TEMPSUMMON_CORPSE_DESPAWN, 600000000);
+                if (creature)
```

Comment:
```suggestion
                
                if (Creature* creature = player->SummonCreature(NPC_ANDOROV, -8538.177f, 1486.0956f, 32.39054f, 3.7638654f, TEMPSUMMON_CORPSE_DESPAWN, 600000000))
```

Reply:
Should he never despawn? Then use TEMPSUMMON_MANUAL_DESPAWN instead of the arbitrary huge despawn time

Reply:
He should despawn immediately. I set to the despawn time to 0.